core/src/test/scala/io/finch/route/RouterSpec.scala

```diff
@@ -0,0 +1,172 @@
+  }
+
+  it should "match method" in {
+    Get(route) shouldBe Some(route.tail)
+  }
+
+  it should "match method only once" in {
+    val r = Get / Get
+    r(route) shouldBe None
+  }
+
+  it should "match string" in {
+    val r = Get / "a"
+    r(route) shouldBe Some(route.drop(2))
+  }
+
+  it should "match 2 or more strings" in {
+    val r = Get / "a" / 1 / "b"
+    r(route) shouldBe Some(route.drop(4))
+  }
+
+  it should "not match if one of the routers failed" in {
+    val r = Get / "a" / 2
+    r(route) shouldBe None
+  }
+
+  it should "not match if the method is missing" in {
+    val r = "a" / "b"
+    r(route) shouldBe None
+  }
+
+  it should "be able to skip route tokens" in {
+    val r = * / "a"
+    r(route) shouldBe Some(route.drop(2))
+  }
+
+  it should "match either one or other matcher" in {
+    val r = (Get | Post) / ("a" | "b")
+    r(route) shouldBe Some(route.drop(2))
+  }
+
+  it should "match int" in {
+    val r = Get / "a" / 1
+    r(route) shouldBe Some(route.drop(3))
+  }
+
+  it should "be able to not match int if it's a different value" in {
+    val r = Get / "a" / 2
+    r(route) shouldBe None
+  }
+
+  it should "be able to skip one route tokens" in {
+    val r = Get / *
+    r (route) shouldBe Some(route.drop(2))
+  }
+
+  it should "be able to match the whole route" in {
+    val r1 = Get / * / $
+    val r2 = Get / * / * / * / * / $
+    r1(route) shouldBe None
+    r2(route) shouldBe Some(Nil)
+  }
+
+  it should "support DSL for string and int extractors" in {
+    val r1 = Get / "a" / int / string
+    val r2 = Get / "a" / int("1") / "b" / int("2") / $
+
+    r1(route) shouldBe Some((route.drop(4), /(1, "b")))
+    r2(route) shouldBe Some((Nil, /(1, 2)))
+  }
+
+  it should "support DSL for boolean marchers and extractors" in {
+    val route = List(PathToken("flag"), PathToken("true"))
+    val r1 = "flag" / boolean
+    val r2 = "flag" / true
+
+    r1(route) shouldBe Some((Nil, true))
+    r2(route) shouldBe Some(Nil)
+  }
+
+  it should "be implicitly converted into a service" in {
+    def echo(s: String) = new Service[HttpRequest, String] {
+      def apply(request: HttpRequest) = s.toFuture
+    }
+
+    val service: Endpoint[HttpRequest, String] = Get / string /> echo
+    Await.result(service(httpx.Request("/foo"))) shouldBe "foo"
+  }
+
+  it should "be composable as an endpoint" in {
+    val r1 = Get / "a" / int /> { _ + 10 }
+    val r2 = Get / "b" / int / int /> { case a / b => a + b }
+    val r3 = r1 | r2
+
+    r3(route) shouldBe Some((route.drop(3), 11))
+  }
+
+  it should "maps to value" in {
+    val r = Get /> 10
+    r(route) shouldBe Some((route.tail, 10))
+  }
+
+  it should "converts into a string" in {
+    val r1 = Get
+    val r2 = Get / "a" / true / 1
+    val r3 = Get / ("a" | "b") / int / long / string
+    val r4 = Get / string("name") / int("id") / boolean("flag") / "foo"
+    val r5 = Post / * / $
+
+    r1.toString shouldBe "GET "
```

Comment:
I don't think `GET` is a valid route for an http server, I think the minimum acceptable would be `GET /` which means we probably shouldn't be able to declare a route that consists of only a method.